iroh-net/src/dns.rs

```diff
@@ -97,10 +100,69 @@ pub trait ResolverExt {
         host: N,
         timeout: Duration,
     ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
+
+    /// Looks up node info by DNS name.
+    fn lookup_by_domain(&self, name: &str) -> impl Future<Output = Result<NodeAddr>>;
```

Comment:
fwiw I find `by_domain` a bit odd here, since it is a hostname you're looking up.  `by_name` would read a little more logical here.  but feel free to disagree and leave it as is.

Reply:
Additionally it caught me out that this returned `NodeAddr`.  I only noticed when seeing the impl (which also contains a more extensive doc comment, maybe copy that here?).  Since the `lookup_` prefix is already used by hickory to return `IpAddr`, maybe we could use another prefix?  Like maybe `node_by_`?

Reply:
`by_name` feels more intuitive to me, `lookup` doesn't bother me because it's still a lookup and it still returns ip addresses, just conveniently bundled up with the rest of the required info, so I'd leave it at that. If you feel strongly about it we can discuss it further ofc. In the meantime I've done just the first change in 9be4c1c